src/appsignal/config.py

```diff
@@ -96,100 +44,218 @@ class Options(TypedDict, total=False):
     working_directory_path: Optional[str]
 
 
-CA_FILE_PATH = os.path.join(
-    os.path.dirname(os.path.abspath(__file__)), "resources", "cacert.pem"
-)
-
-DEFAULT_CONFIG = Options(
-    ca_file_path=CA_FILE_PATH,
-    enable_host_metrics=True,
-    enable_nginx_metrics=False,
-    enable_statsd=False,
-    endpoint="https://push.appsignal.com",
-    files_world_accessible=True,
-    send_environment_metadata=True,
-    send_params=True,
-    send_session_data=True,
-)
+class Config:
+    CA_FILE_PATH = os.path.join(
+        os.path.dirname(os.path.abspath(__file__)), "resources", "cacert.pem"
+    )
+    DEFAULT_CONFIG = Options(
+        ca_file_path=CA_FILE_PATH,
+        enable_host_metrics=True,
+        enable_nginx_metrics=False,
+        enable_statsd=False,
+        endpoint="https://push.appsignal.com",
+        files_world_accessible=True,
+        send_environment_metadata=True,
+        send_params=True,
+        send_session_data=True,
+    )
 
+    DefaultInstrumentation = Literal[
+        "opentelemetry.instrumentation.celery",
+        "opentelemetry.instrumentation.django",
+        "opentelemetry.instrumentation.jinja2",
+        "opentelemetry.instrumentation.psycopg2",
+        "opentelemetry.instrumentation.redis",
+        "opentelemetry.instrumentation.requests",
+    ]
+    DEFAULT_INSTRUMENTATIONS = cast(
+        list[DefaultInstrumentation], list(get_args(DefaultInstrumentation))
+    )
 
-def log_file_path(config: Options) -> Optional[str]:
-    path = config.get("log_path")
+    def __init__(self, options: Optional[Options]):
```

Comment:
Confusingly, the type `Optional` means that it can be `None` (like `Option` in Rust) but as you've probably noticed, it doesn't make the parameter itself is optional. For that, like in Ruby, you need to give it a default value, so that you don't need to pass `None` explicitly:
```suggestion
    def __init__(self, options: Optional[Options] = None):
```